HW05_Xingruo_Zhang.Rmd

```diff
@@ -0,0 +1,121 @@
+---
+title: "HW05_Xingruo_Zhang"
+author: "Summer Zhang"
+date: "8/2/2020"
+output: github_document
+---
+
+
+```{r, message = FALSE}
+library(tidyverse)
+library(readr)
+library(knitr)
+library(stringr)
+
+dad_mom <- read_delim("dad_mom.txt", delim = "\t")
+
+auc_1 <- read_csv("CTRP_files/AUC_1.csv")
+auc_2 <- read_csv("CTRP_files/AUC_2.csv")
+cancer_cell_line_info <- read_csv("CTRP_files/cancer_cell_line_info.csv")
+compound_info <- read_csv("CTRP_files/compound_info.csv")
+Experiment_info <- read_csv("CTRP_files/Experiment_info.csv")
+```
+
+## 1. Tidying the dad_mom file
+```{r}
+dad_mom2 <- dad_mom %>% 
+  # Combine parents' names into one column instead of two
+  gather(key = "parent", value = "name_parent", name_dad, name_mom) %>% 
+  # Correct values in the parent column
+  mutate(parent = ifelse(parent == "name_dad", "dad", "mom")) %>% 
+  # Assign correct income to each parent
+  mutate("income_parent" = ifelse(parent == "dad", income_dad, income_mom)) %>% 
+  # Drop the original income columns
+  select(-c(income_dad,income_mom)) %>% 
+  # Arrange by family ID
+  arrange(fam_id)
+
+dad_mom2
+```
+
+## 2. Joining together CTRP data
+```{r}
+# Inspect the data
+#head(auc_1)
+#head(auc_2)
+#head(cancer_cell_line_info)
+#head(compound_info)
+#head(Experiment_info)
+```
+
+
+```{r}
+auc <- rbind(auc_1, auc_2)
+
+cancer1 <- auc %>% 
+  full_join(compound_info, by = "master_cpd_id")
+
+colnames(Experiment_info)[1] <- "experiment_id"
+
+cancer2 <- cancer1 %>% 
+  full_join(Experiment_info, by = "experiment_id")
+
+cancer_final <- cancer2 %>% 
+  full_join(cancer_cell_line_info, by = "master_ccl_id")
+```
+
+* Which cancer type has the lowest AUC values to the compound "vorinostat"?
+```{r}
+vorinostat_auc <- cancer_final %>% 
+  filter(cpd_name == "vorinostat") %>% 
+  group_by(cancer_type) %>% 
+  summarise(mean_AUC = mean(area_under_curve))
+
+vorinostat_auc$cancer_type[is.na(vorinostat_auc$cancer_type)] <- "NA"
+
+p1 <- ggplot(vorinostat_auc, aes(x = reorder(cancer_type, -mean_AUC), y = mean_AUC)) +
+  geom_col() + 
+  theme(axis.text.x = element_text(angle = 90, vjust=0.5, hjust=1)) +
+  labs(x = 'cancer type', y = "mean AUC" )
+
+print(p1)
```

Comment:
I am not sure why but I couldn't see the plot in the markdown file. But I can tell from the code that the information is well presented in the plot!